lnwallet/channel.go

```diff
@@ -7346,7 +7346,12 @@ func (lc *LightningChannel) MaxFeeRate(maxAllocation float64) chainfee.SatPerKWe
 	// baseBalance is the maximum amount available for us to spend on fees.
 	baseBalance := availableBalance.ToSatoshis() + oldFee
 
+	// In case our local channel balance is drained, we make sure we do not
+	// decrease the fee rate below the old fee rate. This could lead to a
+	// scenario where we lower the commitment fee rate as low as the fee
+	// floor although current fee rates are way higher.
 	maxFee := float64(baseBalance) * maxAllocation
+	maxFee = math.Max(maxFee, float64(oldFee))
```

Comment:
A bit confused here - so this means our fee rate can only go up not down? And if we don't have this local balance, how can we cover the fees here?

Reply:
exactly we do not want to lower the max_feerate lower than the old_fee, because this could lead to a situation where we go all the way to 253 sats/kw, we want to decrease the feerate here: https://github.com/lightningnetwork/lnd/blob/master/lnwallet/channel.go#L8456-L8458, basically we want to either have a 3 block sample_feerate which is lower than the old_feerate before we allow a lower feeupdate, otherwise it makes not real sense to lower the MaxFeeRate.

Reply:
In that case I think we should compare it against 3 block sample feerate instead of old feerate, otherwise if the fee spikes for a while then goes down, we'd still use the higher feerate as the max feerate although the current mempool is less congested. Plus this would bypass the value `maxAllocation` configured by the users?

Reply:
Hmm not sure if I understand the detail of your proposal could you elaborate ?

I think binding the maxAllocation to our current amount we have might cause this strange logic problem.

What about refering the `maxAllocation` do the ChannelCapacity, I think that makes more sense because as a channelopener you will have most of the balance on your side anyways at the beginning and are willing to pay this max amount anyways. Now when the channel becomes depeleted the maxFeeRate is always (`localbalance+oldfee`) but if we have a lot of balance on our side it would be capped at the (`chancapacity *MaxAllocation`) ?

Reply:
We would be protected against big feespikes, not allocating more that our initial willingess to pay, but if the channel is depleted we make sure we do not decrease our MaxFee the more our channel gets depleted hence reducing the local fee of the commitment all the time, that also not fair to the peer imo which now has all the balance but a crappy feerate.

Reply:
Yes I think we can go with the current approach, but I think the FeeRate for the Commitment Tx is really not an easy to solve problem. So I think making sure that the Commitment has always the Relay Fee of its peer is a very important PR which should be prioritized.